File: JAVA/Practica8/PROBLEMA_DIAMANTE/D.java
package JAVA.Practica8.PROBLEMA_DIAMANTE;

public class D extends A {
    public int y;
    public D(int x, int y, int z) {
        super(x, z);
        this.y = y;
    }
    public void incrementaXYZ() {
        this.x += 1;
        this.y += 1;
        this.z += 1;
    }
}

File: PYTHON/Practica8/PROBLEMA_DIAMANTE/D.py
from A import A
from B import B

class D(A, B):
    def __init__(self, x, y, z):
        A.__init__(self, x, z)
        B.__init__(self, y, z)
    def incrementaXYZ(self):
        self.x += 1
        self.y += 1
        self.z += 1


